### DISCUSSION TITLE
Cannot update due to `z` plugin?

### DISCUSSION THREAD

**Discussion by szageda** [ohmyzsh/ohmyzsh#11310] [Help] [answered]
Hi,

I haven't been able to update Oh My Zsh due to an error caused by the `z` plugin.

This is the error I get when I attempt to update:
```bash
$ omz update
plugins/z/Makefile: needs merge
plugins/z/README: needs merge
plugins/z/z.1: needs merge
plugins/z/z.sh: needs merge
error: you need to resolve your current index first
```
I've attached the output of [git status ](https://github.com/ohmyzsh/ohmyzsh/files/9923597/git-status.txt) – it instructs me to run `git pull`, but it results in:
```bash
$ git pull
error: Pulling is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to make resolution and make a commit.
fatal: Exiting because of an unresolved conflict.
```

Even though `git status` says
```bash
..
Unmerged paths:
 (use "git restore --staged <file>..." to unstage)
 (use "git add/rm <file>..." as appropriate to make resolution)
        deleted by us:  plugins/z/Makefile
        deleted by us:  plugins/z/README
        deleted by us:  plugins/z/z.1
        deleted by us:  plugins/z/z.zsh
```
The directory and files are present:
```bash
$ ls -lsr ~/.config/oh-my-zsh/plugins
total 1228
4 drwxr-xr-x. 4 shancy shancy 4096 Oct 22 16:18 zsh-navigation-tools
4 drwxr-xr-x. 2 shancy shancy 4096 Oct 22 16:18 zsh-interactive-cd
4 drwxr-xr-x. 2 shancy shancy 4096 Oct 22 16:18 zoxide
4 drwxr-xr-x. 2 shancy shancy 4096 Oct 22 16:18 zeus
4 drwxr-xr-x. 2 shancy shancy 4096 Oct 22 16:18 zbell
4 drwxr-xr-x. 2 shancy shancy 4096 Oct 22 16:18 z
4 drwxr-xr-x. 2 shancy shancy 4096 Oct 22 16:18 yum
..
$ ls -lsr ~/.config/oh-my-zsh/plugins/z
total 96
12 -rwxr-xr-x. 1 shancy shancy  9359 Oct 22 16:18 z.sh
32 -rwxr-xr-x. 1 shancy shancy 30170 Oct 22 16:18 z.plugin.zsh
 8 -rwxr-xr-x. 1 shancy shancy  5086 Oct 22 16:18 z.1
 4 -rw-r--r--. 1 shancy shancy  2647 Oct 22 16:18 _z
 4 -rwxr-xr-x. 1 shancy shancy   713 Oct 22 16:18 README.md
 8 -rwxr-xr-x. 1 shancy shancy  5430 Oct 22 16:18 README
20 -rw-r--r--. 1 shancy shancy 17702 Oct 22 16:18 MANUAL.md
 4 -rwxr-xr-x. 1 shancy shancy    59 Oct 22 16:18 Makefile
 4 -rw-r--r--. 1 shancy shancy  1078 Oct 22 16:18 LICENSE
```

I've tried running a couple of git commands to fix the issue, but to no avail: `git fetch origin`, `git pull origin`, `git submodule update --remote --recursive`. I haven't tried running `git restore --staged <file>` and `git add/rm <file>` commands as I'm not entirely sure what they do and wouldn't like to make the situation worse than it is.

Is there a way to fix this without having to delete the Oh My Zsh directory then cloning from GitHub again?

Thanks :pray:

-------
### Specs:
- **OS:** Fedora Linux 36 (KDE Plasma) x86_64
- **Kernel:** 6.0.5-200.fc36.x86_64
- **Shell:** zsh 5.8.1
- **Oh My Zsh version:** master (a7d910c)
- **Terminal:** Konsole (tmux)

**Comment by carlosala** [comment, marked as answer]
You can check what to do [here](https://github.com/ohmyzsh/ohmyzsh/issues/11058#issuecomment-1190024382).
Thanks for your report, and have a nice week ahead! 😉
